Purchases/Purchasing/StoreKitAbstractions/StoreProductDiscount.swift

```diff
@@ -233,4 +233,9 @@ extension StoreProductDiscount.DiscountType {
 
 extension StoreProductDiscount.PaymentMode: Encodable {}
 
-extension StoreProductDiscount: Identifiable {}
+extension StoreProductDiscount: Identifiable {
+
+    /// The stable identity of the entity associated with this instance.
+    public var id: String? { return self.offerIdentifier }
```

Comment:
This is weird. What happens if it's nil?

Reply:
Yeah it's weird, but it works™.
The idea is that 2 discounts with `nil` identifier would have the same id. So in SwiftUI, it means you wouldn't be able to have both on a list at the same time.

I've added a note in the PR now to document that this is only `nil` for SK1 discounts pre iOS 12.2.

Reply:
could we merge together the fields used for `isEqual`? I'm slightly worried that on older OS versions, you'd be getting multiple discounts for different products with the same null id

Reply:
Good idea. Done.